tests/ltp/patches/fix_mkdir_mkdir02.patch

```diff
@@ -0,0 +1,61 @@
+diff --git a/testcases/kernel/syscalls/mkdir/mkdir02.c b/testcases/kernel/syscalls/mkdir/mkdir02.c
+index f09e3c33d..50dbcd2e5 100644
+--- a/testcases/kernel/syscalls/mkdir/mkdir02.c
++++ b/testcases/kernel/syscalls/mkdir/mkdir02.c
+@@ -17,6 +17,8 @@
+ #include <unistd.h>
+ #include <stdlib.h>
+ #include "tst_test.h"
++#include <sys/prctl.h>
++#include <linux/securebits.h>
+ 
+ #define TESTDIR1	"testdir1"
+ #define TESTDIR2	"testdir1/testdir2"
+@@ -54,10 +56,8 @@ static void verify_mkdir(void)
+ 		if (!fail)
+ 			tst_res(TPASS, "New dir inherited GID and S_ISGID");
+ 
+-		exit(0);
+ 	}
+ 
+-	tst_reap_children();
+ 	SAFE_RMDIR(TESTDIR2);
+ }
+ 
+@@ -67,6 +67,16 @@ static void setup(void)
+ 	struct passwd *pw;
+ 	struct stat buf;
+ 	pid_t pid;
++	int usr_secbits;
++
++	// set the flag "SECBIT_NO_SETUID_FIXUP" to retain the
```

Comment:
It's not clear why this patch is needed to make mkdir work. Could you add more details?
Also, please use `if(getuid() == 0)` since your intention is not comparing a boolean.

Reply:
Modified the expression in if condition. Details about the patch is added in patch description.